scality_glance_store/store.py

```diff
@@ -117,6 +118,64 @@ def __init__(self, conf):
     def get_schemes():
         return (SCALITY_SCHEME,)
 
+    def _get_range_headers(self, location, offset=0, chunk_size=None):
+        """
+        Calculate range headers for partial object retrieval.
+        """
+        # Obtain object size by a HEAD request. This is to ensure compatibiliy
+        # with versions of sproxyd which has limited support for implicit
+        # ranges in the HTTP Range header.
+        size = self.get_size(location)
+
+        if chunk_size is not None:
+            # Explicit range request.
+            if chunk_size <= 0:
```

Comment:
Could you perform input validation (the parts that don't depend on `size`) before performing any side-effects (through `get_size`)?